src/modules/graphql/resolvers/permissionRules.ts

```diff
@@ -0,0 +1,80 @@
+import { UserRole} from '../../../models/user.db.model';
+import { UserTokenProfile } from '../../../models/user-token-profile.model';
+
+export const NO_PERMISSION = 'not permissions to execute command';
+
+export enum DBOperation {
+    CREATE = 'CREATE',
+    READ = 'READ',
+    UPDATE = 'UPDATE',
+    DELETE = 'DELETE',
+    ANY = 'ANY',
+}
+
+export enum Asset {
+    STUDENT = 'STUDENT',
+    CLASS = 'CLASS',
+    LESSON = 'LESSON',
+    MAP = 'MAP',
+    USER = 'USER',
+    ANY = 'ANY',
+}
+
+export enum Permission {
+    ALLOW = 'ALLOW',
+    FORBID = 'FORBID',
+    OWN = 'OWN',
+}
+
+export interface PermissionRule {
+    operation: DBOperation;
+    asset: Asset;
+    role: UserRole;
+    permission: Permission;
+}
+
+export const TEACHER_PERMISSION_RULES: PermissionRule[] = [
+    // Asset.STUDENT
+    {operation: DBOperation.CREATE, asset: Asset.STUDENT, role: UserRole.TEACHER, permission: Permission.FORBID},
+    {operation: DBOperation.READ, asset: Asset.STUDENT, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.UPDATE, asset: Asset.STUDENT, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.DELETE, asset: Asset.STUDENT, role: UserRole.TEACHER, permission: Permission.FORBID},
+    // Asset.LESSON
+    {operation: DBOperation.CREATE, asset: Asset.LESSON, role: UserRole.TEACHER, permission: Permission.FORBID},
+    {operation: DBOperation.READ, asset: Asset.LESSON, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.UPDATE, asset: Asset.LESSON, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.DELETE, asset: Asset.LESSON, role: UserRole.TEACHER, permission: Permission.FORBID},
+    // Asset.CLASS
+    {operation: DBOperation.CREATE, asset: Asset.CLASS, role: UserRole.TEACHER, permission: Permission.FORBID},
+    {operation: DBOperation.READ, asset: Asset.CLASS, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.UPDATE, asset: Asset.CLASS, role: UserRole.TEACHER, permission: Permission.OWN},
+    {operation: DBOperation.DELETE, asset: Asset.CLASS, role: UserRole.TEACHER, permission: Permission.FORBID},
+];
+
+export class Permissions {
+
+    static getPermission(asset: Asset, op: DBOperation, user: UserTokenProfile): Permission {
+        if (user.role === UserRole.PRINCIPLE) {
```

Comment:
i would consider to use strategy pattern here to simplify code 
https://codeburst.io/implementing-strategy-pattern-on-typescript-b74c447da37b